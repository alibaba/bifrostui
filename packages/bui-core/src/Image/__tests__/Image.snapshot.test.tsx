import { snapshotTest } from 'testing';

describe('Image snapshot', () => {
  snapshotTest('Image');
});
