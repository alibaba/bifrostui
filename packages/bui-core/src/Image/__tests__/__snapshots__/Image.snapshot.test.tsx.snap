// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image snapshot Image demo snapshot 0 1`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    className="bui-image"
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg"
      style={
        Object {
          "objectFit": "fill",
        }
      }
    />
  </div>
</div>
`;

exports[`Image snapshot Image demo snapshot 0 2`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "row",
      "--flex-wrap": "wrap",
      "--justify-content": "center",
      "--spacing": "8px",
    }
  }
>
  <div
    className="bui-image"
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      alt="contain"
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg"
      style={
        Object {
          "objectFit": "contain",
        }
      }
    />
  </div>
  <div
    className="bui-image"
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      alt="cover"
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg"
      style={
        Object {
          "objectFit": "cover",
        }
      }
    />
  </div>
  <div
    className="bui-image"
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      alt="fill"
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg"
      style={
        Object {
          "objectFit": "fill",
        }
      }
    />
  </div>
  <div
    className="bui-image"
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      alt="none"
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg"
      style={
        Object {
          "objectFit": "none",
        }
      }
    />
  </div>
  <div
    className="bui-image"
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      alt="scale-down"
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg"
      style={
        Object {
          "objectFit": "scale-down",
        }
      }
    />
  </div>
</div>
`;

exports[`Image snapshot Image demo snapshot 0 3`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "row",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "8px",
    }
  }
>
  <div
    className="bui-image"
    style={
      Object {
        "borderRadius": 4,
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg"
      style={
        Object {
          "objectFit": "cover",
        }
      }
    />
  </div>
  <div
    className="bui-image"
    style={
      Object {
        "borderRadius": 8,
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg"
      style={
        Object {
          "objectFit": "cover",
        }
      }
    />
  </div>
  <div
    className="bui-image"
    style={
      Object {
        "borderRadius": 32,
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg"
      style={
        Object {
          "objectFit": "cover",
        }
      }
    />
  </div>
</div>
`;

exports[`Image snapshot Image demo snapshot 0 4`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "row",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "8px",
    }
  }
>
  <div
    className="bui-image"
    style={
      Object {
        "borderRadius": 4,
        "height": 100,
        "width": 100,
      }
    }
  >
    <div
      className="bui-image-default-icon-wrap"
    >
      <div
        className="bui-image-default-icon-item"
      />
    </div>
    <img
      className="bui-image-img"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://localhost:3000/failed"
      style={
        Object {
          "objectFit": "cover",
        }
      }
    />
  </div>
  <div
    className="bui-image"
    style={
      Object {
        "borderRadius": 4,
        "height": 100,
        "width": 100,
      }
    }
  >
    <img
      className="bui-image-img bui-image-loaded"
      loading="eager"
      onError={[Function]}
      onLoad={[Function]}
      src="https://localhost:3000/failed"
      style={
        Object {
          "objectFit": "cover",
        }
      }
    />
  </div>
</div>
`;

exports[`Image snapshot Image demo snapshot 0 5`] = `
Array [
  <button
    className="bui-btn bui-btn-outlined bui-btn-medium bui-btn-default"
    onClick={[Function]}
    type="button"
  >
    重新加载
  </button>,
  <div
    className="bui-stack"
    style={
      Object {
        "--align-items": "center",
        "--flex-direction": "row",
        "--flex-wrap": "nowrap",
        "--justify-content": "center",
        "--spacing": "8px",
      }
    }
  >
    <div
      className="bui-image"
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    >
      <img
        className="bui-image-img bui-image-loaded"
        loading="eager"
        onError={[Function]}
        onLoad={[Function]}
        src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg?t=0"
        style={
          Object {
            "objectFit": "fill",
          }
        }
      />
    </div>
    <div
      className="bui-image"
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    >
      <div
        className="bui-image-default-icon-wrap"
      >
        <div
          className="bui-image-default-icon-item"
        />
      </div>
      <img
        className="bui-image-img"
        loading="eager"
        onError={[Function]}
        onLoad={[Function]}
        src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg?t=0"
        style={
          Object {
            "objectFit": "fill",
          }
        }
      />
    </div>
    <div
      className="bui-image"
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    >
      <div
        className="bui-image-default-icon-wrap"
      >
        <div
          style={
            Object {
              "background": "red",
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <img
        className="bui-image-img"
        loading="eager"
        onError={[Function]}
        onLoad={[Function]}
        src="https://gw.alicdn.com/i2/O1CN01D7yqW229UZMB5eh00_!!6000000008071-0-alipicbeacon.jpg?t=0"
        style={
          Object {
            "objectFit": "fill",
          }
        }
      />
    </div>
  </div>,
]
`;
